File: backend/src/main/kotlin/org/icpclive/events/codeforces/api/data/CFProblemResult.java
package org.icpclive.events.codeforces.api.data;

/**
 * @author egor@egork.net
 */
public class CFProblemResult {
    public double points;
    public int penalty;
    public int rejectedAttemptCount;
    public CFProblemResultType type;
    public long bestSubmissionTimeSeconds;

    public enum CFProblemResultType {
        PRELIMINARY, FINAL
    }
}

File: backend/src/main/kotlin/org/icpclive/cds/codeforces/api/data/CFProblemResult.kt
package org.icpclive.cds.codeforces.api.data

import kotlinx.serialization.*

enum class CFProblemResultType {
    PRELIMINARY, FINAL
}

@Serializable
data class CFProblemResult(
    val points: Double,
    val penalty: Int? = null,
    val rejectedAttemptCount: Int,
    val type: CFProblemResultType,
    val bestSubmissionTimeSeconds: Long? = null,
)